src/flyte/syncify/_api.py

```diff
@@ -150,6 +150,17 @@ async def iterate_in_loop(self, async_gen: AsyncIterator[R_co]) -> AsyncIterator
                 yield r
             return
 
+        # Install the gRPC error handler on the caller's event loop as well.
+        # This is needed because gRPC's async polling events may be delivered to
+        # the caller's loop (e.g., FastAPI's event loop) when using .aio().
+        import flyte.errors
+
+        try:
+            caller_loop = asyncio.get_running_loop()
```

Comment:
I think we should not do it again and again, we should only do it once really, i would put a variable block

Reply:
like have a thread local variable that can be used to check if things have been set